spec/recurly/subscription_spec.rb

```diff
@@ -76,11 +76,15 @@
           ramp_intervals: [
             SubscriptionRampInterval.new(
               starting_billing_cycle: 1,
-              unit_amount_in_cents: 1000
+              unit_amount_in_cents: 1000,
```

Comment:
I tried to implement similar testing to what is being done [here](https://github.com/recurly/recurly-client-python/pull/655/files), but it's not really the same because the Python version is making a request based on the fixture file while the Ruby client is doing the test set up in the before block and just using the fixture file for comparison. Is there another way to test the addition of these fields, or is this enough?

Reply:
So your read is right, putting the test here he a little odd, since this test is basically testing that a Subscription with ramps serializes to an XML format that could be used to say, make a POST request.(notice how remaining_billing_cycles is not in the XML, since it is not something included in a request, only in a response)

The thing we're implementing is 'Can read starting_on and ending_on from an xml response`  I think a correct test would be that you can read the attributes from an XML response, so something like the following

``` 
sub - Subscription.from_xml(get_raw_xml 'fixture-you made')

expect(sub.ramp_intervals.first.starting_on).to eq(date_from_fixture)
expect(sub.ramp_intervals.first.ending_on).to eq(date_from_fixture)

expect(sub.ramp_intervals.second.starting_on).to eq(date_from_fixture)
expect(sub.ramp_intervals.secong.ending_on).to eq(nil)
```

That said.. it seems a little redundant to test these, since we already have tests for resource.rb. I do think I would lean towards a test like this though, since these changes don't require any PG changes. Im pretty sure if you take those two lines around `starting_on` and `ending_on` out of `define_attribute_methods` the test would fail. 

I think I would write this test. While doing so, I would also put an expectation for `remaining_billing_cycles` and add it to the fixture and see if it fails without that method in `define_attribute_methods` `

Reply:
I reverted the fixture from the original test back to what it was and created a new fixture file for this one. Confirmed that removing `starting_on` and `ending_on` out of `define_attribute_methods` causes the test to fail. Added expectations for `starting_billing_cycle` and `unit_amount_in_cents` that also fail if those attributes are removed.